SonarQube.TeamBuild.PreProcessor/Roslyn/RoslynAnalyzerProvider.cs

```diff
@@ -25,68 +25,47 @@ public class RoslynAnalyzerProvider
         public const string CSharpPluginKey = "csharp";
         public const string CSharpRepositoryKey = "csharp";
 
-        private readonly ISonarQubeServer server;
-        private readonly TeamBuildSettings settings;
-        private readonly string projectKey;
         private readonly ILogger logger;
-
+        private ISonarQubeServer server;
+        private TeamBuildSettings settings;
+        private string projectKey;
+        
         #region Public methods
 
-        /// Sets up the client to run the Roslyn analyzers as part of the build
-        /// i.e. creates the Roslyn ruleset and provisions the analyzer assemblies
-        /// and rule parameter files
-        /// </summary>
-        public static CompilerAnalyzerConfig SetupAnalyzers(ISonarQubeServer server, TeamBuildSettings settings, string projectKey, ILogger logger)
+        public RoslynAnalyzerProvider(ILogger logger)
         {
-            if (server == null)
+            if (logger == null)
+            {
+                throw new ArgumentNullException("logger");
+            }
+            this.logger = logger;
+        }
+
+        public CompilerAnalyzerConfig SetupAnalyzers(ISonarQubeServer sqServer, TeamBuildSettings teamBuildSettings, string sqProjectKey)
```

Comment:
Changed the method to be non-static, which meant I had to rename the parameters to stop static analysis complaints about them hiding fields of the same name.